9-regular-expressions/01-regexp-introduction/article.md

```diff
@@ -1,91 +1,91 @@
-# Patterns and flags
+# الگو ها و پرچم ها (Patterns and flags)
 
-Regular expressions are patterns that provide a powerful way to search and replace in text.
+عبارات باقاعده (Regular expressions) الگوهایی هستند که روشی قدرتمند برای جستجو و جایگزینی در متن ارائه می دهند.
 
-In JavaScript, they are available via the [RegExp](mdn:js/RegExp) object, as well as being integrated in methods of strings.
+در جاوا اسکریپت، آنها از طریق شیء [RegExp](mdn:js/RegExp) استفاده می شوند و همچنین به خوبی با متد های رشته ها ادغام می شوند.
 
-## Regular Expressions
+## عبارات باقاعده (Regular Expressions)
 
-A regular expression (also "regexp", or just "reg") consists of a *pattern* and optional *flags*.
+عبارت باقاعده (همچنین "regexp"، یا فقط "reg") از یک *الگو(pattern)* و *پرچم(flags)* های اختیاری تشکیل می شوند.
 
-There are two syntaxes that can be used to create a regular expression object.
+دو روش وجود دارد که می توان از آنها برای ایجاد یک شیء عبارت باقاعده استفاده کرد.
 
-The "long" syntax:
+سینتکس "طولانی":
 
 ```js
-regexp = new RegExp("pattern", "flags");
+regexp = new RegExp("pattern(الگو)", "flags(پرچم)");
 ```
 
-And the "short" one, using slashes `"/"`:
+و سینتکس "کوتاه" نیز، استفاده از اسلش `"/"` می باشد.
 
 ```js
-regexp = /pattern/; // no flags
-regexp = /pattern/gmi; // with flags g,m and i (to be covered soon)
+regexp = /pattern/; // بدون پرچم
+regexp = /pattern/gmi; // با پرچم g, m, i (به زودی پوشش داده می شود)
```

Comment:
```suggestion
regexp = /pattern/gmi; // (به زودی گفته می‌شود) i، m، g با پرچم
```
Please take a look at [this](https://github.com/javascript-tutorial/fa.javascript.info#%D8%AA%D8%B1%D8%AC%D9%85%DB%80-%DA%A9%D8%A7%D9%85%D9%86%D8%AA%D9%87%D8%A7-%D8%AF%D8%B1-%DA%A9%D8%AF) to know how to translate comments of the code.